apps/freeablo/fagui/dialogmanager.cpp

```diff
@@ -0,0 +1,271 @@
+#include "dialogmanager.h"
+#include <cassert>
+#include "guimanager.h"
+
+namespace FAGui
+{
+    DialogLineData& DialogLineData::setAction(std::function<void()> actionArg)
+    {
+        action = actionArg;
+        return *this;
+    }
+
+    DialogLineData& DialogData::text_lines(const std::vector<std::string>& texts, TextColor color, bool alignCenter)
+    {
+        auto& ret = mLines[mLastLine];
+        for(auto& text : texts)
+        {
+            mLines[mLastLine].text = text;
+            mLines[mLastLine].color = color;
+            mLines[mLastLine].isSeparator = false;
+            mLines[mLastLine].alignCenter = alignCenter;
+            ++mLastLine;
+        }
+        skip_line();
+        return ret;
+    }
+
+    void DialogData::skip_line(int cnt)
+    {
+        mLastLine += cnt;
+    }
+
+    void DialogData::separator()
+    {
+        mLines[mLastLine].isSeparator = true;
+        ++mLastLine;
+        skip_line();
+    }
+
+    void DialogData::header(const std::vector<std::string>& text)
+    {
+        if(text.size() == 1)
+        {
+            skip_line();
+            text_lines({text.front()}, TextColor::golden);
+            skip_line();
+        }
+        else
+        {
+            for(auto& line : text)
+            {
+                text_lines({line}, TextColor::golden);
+            }
+        }
+        separator();
+        skip_line();
+    }
+
+    int DialogData::selectedLine()
+    {
+        if(mSelectedLine == -1)
+        {
+            auto it = std::find_if(mLines.begin(), mLines.end(),
+                                   [](const DialogLineData& data) { return !!data.action; });
+            if(it != mLines.end())
+                return mSelectedLine = it - mLines.begin();
+            assert(false);
+        }
+        return mSelectedLine;
+    }
+
+    DialogManager::DialogManager(GuiManager& gui_manager): mGuiManager(gui_manager)
+    {
+    }
+
+    void DialogManager::talkOgden()
+    {
+        DialogData d;
+        d.header({"Welcome to the", "Rising Sun"});
```

Comment:
I'd prefer not to have this text in here. If we're to have placeholder text, just some lorem ipsum would be better

Reply:
Sorry, I don't quite get that. You mean I should extract strings from exe? But before that it was in writtine in .rml all the same, what's the difference?

Reply:
Yeah, I didn't know it was written in the rml before :s
IMO it's better to have placeholder text so we know it's not implemented, than to have the real text and forget it's hardcoded instead of loaded properly.